Goal: ??
Depends: 008_login_more_LOG_UNKFAIL_ENAB

Notes:
 * I've not found a related entry in the changelog.
 * This patch looks strang to me. It adds #ifndef USE_PAM in section already
   enclosed by either #ifdef USE_PAM or #ifndef USE_PAM.

   IMHO, it should ignored or rewritten.

Index: shadow-4.0.3/src/login.c
===================================================================
--- shadow-4.0.3.orig/src/login.c	2005-05-29 00:03:47.374189000 +0200
+++ shadow-4.0.3/src/login.c	2005-05-29 00:03:58.704189000 +0200
@@ -809,6 +809,10 @@
 			    syslog (LOG_NOTICE,
 				    _("TOO MANY LOGIN TRIES (%d)%s FOR `%s'"),
 				    failcount, fromhost, failent_user);
+#ifndef USE_PAM
+			    if (pwd && getdef_bool("FAILLOG_ENAB"))
+			      failure (pwent.pw_uid, tty, &faillog);
+#endif
 			    fprintf(stderr,
 				    _("Maximum number of tries exceeded (%d)\n"),
 				    failcount);
@@ -826,11 +830,22 @@
 				   pam_strerror (pamh, retcode));
 			    failed = 1;
 			  }
+#ifndef USE_PAM
+			  if (pwd && getdef_bool("FAILLOG_ENAB") &&
+			      ! failcheck (pwent.pw_uid, &faillog, failed)) {
+			    SYSLOG((LOG_CRIT, FAILURE_CNT, failent_user, fromhost));
+			    failed = 1;
+			  }
+#endif
 
 			  if (!failed)
 			    break;
 			  
 			  fprintf(stderr,"Login incorrect\n\n");
+#ifndef USE_PAM
+			  if (pwd && getdef_bool("FAILLOG_ENAB"))
+			    failure (pwent.pw_uid, tty, &faillog);
+#endif
 			  if (getdef_str("FTMP_FILE") != NULL) {
 #if HAVE_UTMPX_H
 			    failent = utxent;
@@ -1047,6 +1062,7 @@
 			failed = 1;
 		}
 #endif
+#ifndef USE_PAM
 		if (pwd && getdef_bool ("FAILLOG_ENAB") &&
 		    !failcheck (pwent.pw_uid, &faillog, failed)) {
 			SYSLOG ((LOG_CRIT, 
@@ -1054,12 +1070,15 @@
 				 username, fromhost));
 			failed = 1;
 		}
+#endif
 		if (!failed)
 			break;
 
+#ifndef USE_PAM
 		/* don't log non-existent users */
 		if (pwd && getdef_bool ("FAILLOG_ENAB"))
 			failure (pwent.pw_uid, tty, &faillog);
+#endif
 		if (getdef_str ("FTMP_FILE") != NULL) {
 			const char *failent_user;
 
